src/test/java/io/zipcoder/casino/GoFishTest.java

```diff
@@ -0,0 +1,277 @@
+        Assert.assertEquals(expectedAtIdx0,  actualAtIdx0);
+        Assert.assertEquals(expectedAtIdx1,  actualAtIdx1);
+        Assert.assertEquals(expectedAtIdx13,  actualAtIdx13);
+        Assert.assertEquals(expectedAtIdx28,  actualAtIdx28);
+        Assert.assertEquals(expectedAtIdx51,  actualAtIdx51);
+    }
+
+    @Test
+    public void testShuffleCards() {
+        // Given
+        go.createNewDeck();
+        ArrayList<String> unshuffledDeck = new ArrayList<String>(go.getDeck());
+
+        // When
+        go.shuffleCards();
+        ArrayList<String> shuffledDeck = go.getDeck();
+
+        // Then
+        Assert.assertNotEquals(unshuffledDeck, shuffledDeck);
+    }
+
+    @Test
+    public void testDealCards() {
+        // Given
+        go.createNewDeck();
+        ArrayList<String> actualDeck = go.getDeck();
+
+        // When
+        go.dealCards(5);
+        ArrayList<String> actualPerson1 = go.getPlayersHand();
+        ArrayList<String> actualPerson2 = go.getDealersHand();
+
+        // Then
+        Assert.assertEquals(5, actualPerson1.size());
+        Assert.assertEquals(5, actualPerson2.size());
+        Assert.assertEquals(42, actualDeck.size());
+    }
+
+    @Test
+    public void testGetRankOnCard() {
+        // Given
+        String given = "2 of Diamonds";
+        String expected = "2";
+
+        String givenCase2 = "Ace of Hearts";
+        String expectedCase2 = "Ace";
+
+        String givenCase3 = "10 of Clubs";
+        String expectedCase3 = "10";
+
+        // When
+        String actual = go.getRankOnCard(given);
+        String actualCase2 = go.getRankOnCard(givenCase2);
+        String actualCase3 = go.getRankOnCard(givenCase3);
+
+        // Then
+        Assert.assertEquals(expected, actual);
+        Assert.assertEquals(expectedCase2, actualCase2);
+        Assert.assertEquals(expectedCase3, actualCase3);
+    }
+
+    @Test
+    public void testGetValueOnCard() {
+        // Given
+        int expectedRank = 10;
+        int expectedRank2 = 9;
+        String inputRank = "Ace";
+        String inputRank2 = "9";
+
+        // When
+        int actualRank = go.getValueOfCard(inputRank);
+        int actualRank2 = go.getValueOfCard(inputRank2);
+
+        Assert.assertEquals(expectedRank, actualRank);
+        Assert.assertEquals(expectedRank2, actualRank2);
+    }
+
+    @Test
+    public void testDisplayPlayersHand() {
+        // Given
+        ArrayList<String> playersHand = new ArrayList<String>(Arrays.asList("2 of Diamonds", "Queen of Hearts", "3 of Spades", "4 of Clubs", "7 of Hearts"));
+        go.setPlayersHand(playersHand);
+        String expected = "2 of Diamonds\nQueen of Hearts\n3 of Spades\n4 of Clubs\n7 of Hearts\n";
+
+        // When
+        String actual = go.displayPlayersHand(go.getPlayersHand());
+
+        // Then
+        Assert.assertEquals(expected, String.valueOf(actual));
+    }
+
+    @Test
+    public void TestGetValueOfCard() {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.